### ISSUE TITLE
Idea: Better handling of RTSP streams

### ISSUE THREAD

**Issue by Flight777** [briis/unifiprotect#238]
Reading the change log on the latest release triggered me to share this :)

Recently I've been using this component: https://github.com/AlexxIT/WebRTC to enable realtime streaming of my Unifi RTSP feeds.

Might be something to embed in the Unifi Protect integration itself as well, instead of the JPEG workaround?

**Comment by bdraco** [comment]
This would need to be integrated into core.  The repo you linked downloads a go binary instead of a native python application which would need to be resolved before it could be integrated into core

**Comment by bdraco** [comment]
https://github.com/aiortc/aiortc and https://github.com/nicotyze/Webrtc-H264Capturer look like they could be useful

**Comment by AngellusMortis** [comment]
I have been doing a lot of thinking about the streaming and everything. It would definitely need to be something integrated into HA core first. 

Also, because Unifi does _not_ allow changing the output encoding, it would also need to be a solution that supports transcoding audio for WebRTC (Unifi Protect outputs in AAC which no browser implementation of WebRTC supports), so the second item looks like it would like not work. It looks like the first one (aiortc) uses ffmpeg under the hood so that could handle the audio transcode.

**Comment by NVader2000** [comment]
Flight777, WebRTC deployment appears to be the only option for HA. However, I believe after version 1.4 stability for RSTP tanked. So much so that I simply removed it from my HA and hence why I am here looking for a better solution. I used both a web browser and the HA Android app on several device and the RSTP streams with have massive delays (e.g. days) or would simply lock up asking for a restart/reconnection. In fact, another user on that post speak to this as well.

My network is not the issue as I stream far more intensive videos (e.g. multiple 4K) to various device without issues. It is either HA, this WebRTC, or UniFi Protect. This integrate appeared to be hopeful, but quickly found that there is no instruction on how to add the RSTP streams from the camera that it detects.

**Comment by stephenjamieson** [comment]
So, now that Home Assistant has webrtc for Nest, does that help here?

**Comment by AngellusMortis** [comment]
No. Nest has WebRTC because nest can do WebRTC. Playing a WebRTC is different than making one. 

Using LL-HLS will get you a near 1 second latency. https://www.home-assistant.io/integrations/stream#ll-hls

**Comment by stephenjamieson** [comment]
Got it, sorry I thought unifi protect uses webrtc for streaming in their ui via the internal api
